rpc/rpc.go

```diff
@@ -1296,7 +1296,11 @@ func (c *Conn) recvPayload(payload rpccp.Payload) (_ capnp.Ptr, locals uintSet,
 		var err error
 		mtab[i], err = c.recvCap(ptab.At(i))
 		if err != nil {
-			releaseList(mtab[:i]).release()
+			// FIXME: is it safe to release these while holding c.lk? In general,
+			// it's possible for Client.Release() to deadlock.
```

Comment:
Note: this was a problem with the previous implementation as well, I just noticed it while doing this. Perhaps we should open an issue to follow up?

Reply:
I'm not familiar enough to comment on the issue itself, but yes, let's definitely sort this one out.

Reply:
Alright, opened an issue. Merging.